server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -385,13 +386,25 @@ private void closeSession() {
         }
 
         @Override
-        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
-            if (cause instanceof SocketException && cause.getMessage().equals("Connection reset")) {
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable t) throws Exception {
+            LOGGER.trace("Exception caught", t);
+            if (t instanceof SocketException && t.getMessage().equals("Connection reset")) {
                 LOGGER.info("Connection reset. Client likely terminated connection");
-                closeSession();
+            } else if (t instanceof DecoderException) {
+                Messages.sendErrorResponse(
+                    channel,
+                    session == null
+                        ? AccessControl.DISABLED
+                        : getAccessControl.apply(session.sessionSettings()),
+                    t.getCause(),
+                    PGError.Severity.FATAL,
+                    false
```

Comment:
What's the reason that we exclude the stacktrace here?

Reply:
The intention was to not leak the stacktrace information to clients which aren't sending valid data. Not sure if this adds value, maybe not really worth the complexity added. wdyt?

Reply:
People could always read the code, so I'm not sure if the stacktrace can leak any additional information.
I'd tend to remove it - just to avoid having people need to reason about when to send vs. not to send.